library/systemd/src/lib/yast2/service_widget.rb

```diff
@@ -0,0 +1,165 @@
+  #       service_widget = ServiceWidget.new(@service_configuration)
+  #       content = VBox(
+  #         ...,
+  #         service_widget.content
+  #       )
+  #       loop do
+  #         input = UI.UserInput
+  #         service_widget.handle_input(input)
+  #         ...
+  #       end
+  #       service_widget.store
+  #     end
+  #
+  #     def write
+  #       service_configuration.write
+  #     end
+  #   end
+  class ServiceWidget
+    include Yast::I18n
+    include Yast::Logger
+    include Yast::UIShortcuts
+    # creates new widget instance for given service configuration
+    # @param service_configuration [Yast2::ServiceConfiguration] configuration holder
+    def initialize(service_configuration)
+      textdomain "base"
+      @service_configuration = service_configuration
+    end
+
+    # gets widget term
+    # @return <Yast::Term>
+    def content
+      # TODO: disabling invalid action and autostart or kick it out?
+      Frame(
+        _("Service Configuration"),
+        VBox(
+          Left(
+            HBox(
+              Label(_("Current status:")),
+              Label(" "),
+              Label(status)
+            )
+          ),
+          Left(action_widget),
+          Left(autostart_widget)
+        )
+      )
+    end
+
+    # handles event to dynamically react on user configuration.
+    # For events that does not happen inside widget it is ignored.
+    # @param event_id [Object] id of UI element that cause event
+    # @return [void]
+    def handle_input(event_id)
+      log.info "handle event #{event_id}"
+
+      nil
+    end
+
+    # Stores current configuration. Should be called always even when going
+    # back so configuration is persistent when going again forward.
+    def store
+      store_action
+      store_autostart
+    end
+
+  private
+
+    attr_reader :service_configuration
+
+    def store_action
+      action = Yast::UI.QueryWidget(Id(:service_widget_action), :CurrentItem)
+      return unless action
+
+      service_configuration.action = action.to_s.sub(/^service_widget_action_/, "").to_sym
+    end
+
+    def store_autostart
+      autostart = Yast::UI.QueryWidget(Id(:service_widget_autostart), :CurrentItem)
+      return unless autostart
+
+      service_configuration.autostart = autostart.to_s.sub(/^service_widget_autostart_/, "").to_sym
+    end
+
+    def status
+      case service_configuration.status
+      # TRANSLATORS: Status of service
+      when :active
+        _("Active")
+      when :inactive
+        # TRANSLATORS: Status of service
+        _("Inactive")
+      when :inconsistent
+        # TRANSLATORS: Status of service
+        _("Partly Active")
+      when :unknown
+        # TRANSLATORS: Status of service
+        _("Unknown")
+      else
+        raise "Unknown status #{service_configuration.status.inspect}"
+      end
+    end
+
+    def action_widget
+      ComboBox(
+        Id(:service_widget_action),
+        "After writting settings:",
```

Comment:
@jreidinger 
1. the correct spelling is "wri**t**ing", with one **t** (because that makes for an open, long syllable: wri-ting, pronounced  [rai-tiŋ], whereas "writ-ting" is a closed, short syllable [ri-tiŋ]. There *are* some sensile rules in English spelling)
2. translation marker missing